### ISSUE TITLE
test suite triggers llvm assertions when llvm compiled with assertions on

### ISSUE THREAD

**Issue by andrewrk** [zig-lang/zig#409]
to reproduce:

 * compile llvm, clang, lld with assertions on
 * build zig against these build artifacts
 * run zig test suite and observe it does not pass

**Comment by andrewrk** [comment]
I have one of them down to a small test case:

```zig
pub fn panic(msg: []const u8) -> noreturn { while (true) {} }

export fn entry() {
    var stack_trace: ElfStackTrace = undefined;

    %%stack_trace.elf.openStream(&stack_trace.self_exe_stream);
}

const ElfStackTrace = struct {
    self_exe_stream: InStream,
    elf: Elf,
};

pub const Elf = struct {
    in_stream: &InStream,
    is_64: bool,
    is_big_endian: bool,
    entry_addr: u64,

    pub fn openStream(elf: &Elf, stream: &InStream) -> %void {
        if (elf.is_64) {
            elf.entry_addr = elf.in_stream.readInt(elf.is_big_endian, u64);
        } else {
            elf.entry_addr = u64(elf.in_stream.readInt(elf.is_big_endian, u32));
        }
    }

};

extern fn foo();

pub const InStream = struct {
    fd: i32,

    pub fn readInt(is: &InStream, is_be: bool, comptime T: type) -> T {
        var x = if (is_be) T(1) else T(2);
        foo();
        return x;
    }
};
```

```
[nix-shell:~/dev/zig/build-llvm-debug]$ ./zig build_obj test.zig --release-fast 
zig: /home/andy/downloads/llvm-4.0.1.src/include/llvm/IR/IRBuilder.h:1776: llvm::Value* llvm::IRBuilder<T, Inserter>::CreateExtractInteger(const llvm::DataLayout&, llvm::Value*, llvm::IntegerType*, uint64_t, const llvm::Twine&) [with T = llvm::TargetFolder; Inserter = llvm::IRBuilderDefaultInserter; uint64_t = long unsigned int]: Assertion `DL.getTypeStoreSize(ExtractedTy) + Offset <= DL.getTypeStoreSize(IntTy) && "Element extends past full value"' failed.
Aborted
```

It's tricky because the assertion is in one of the optimization passes and it's not clear where the problem is. Still looking into it.

**Comment by andrewrk** [comment]
This is fixed in the release candidate of llvm 5.

**Comment by andrewrk** [comment]
Given that there's an llvm bug preventing this  from being fixed I'm going to close this in favor of #410 .

We'll focus on building against llvm 5.0.0 rc1.
